eZ/Publish/Core/Repository/Mapper/ContentDomainMapper.php

```diff
@@ -119,6 +127,17 @@ public function buildContentDomainObject(
 
         $versionInfo = $this->buildVersionInfoDomainObject($spiContent->versionInfo, $prioritizedLanguages);
 
+        $contentInfo = $versionInfo->getContentInfo();
+        $mainLocation = $contentInfo->getMainLocation();
+        if ($mainLocation === null && $this->locationHandler->countAllLocations() > 0) {
```

Comment:
This is not correct. You're counting here all Locations. So if Content A has no Locations at all, but Content B has one main properly set Location, this expression would still be true for Content A. You need to count Locations for a given Content.

Reply:
Oops, I thought this method was receiving an argument and that I have provided one (IDE was not highlighting this), silly mistake